tabular/tests/unittests/resource_allocation/test_gpu_assignment.py

```diff
@@ -0,0 +1,490 @@
+        return None
+
+    @ray.remote
+    def ray_task_report_gpu_info(task_id, assigned_gpu_ids):
+        """
+        Ray remote task that reports GPU information.
+
+        This simulates what happens in _ray_fit() at fold_fitting_strategy.py:428
+        where GPU assignments are used to set CUDA_VISIBLE_DEVICES.
+
+        Parameters
+        ----------
+        task_id : int
+            The task ID for identification
+        assigned_gpu_ids : list
+            The list of GPU IDs assigned to this task
+
+        Returns
+        -------
+        dict
+            Dictionary containing GPU info reported by the task
+        """
+        import os
+        try:
+            import torch
+        except ImportError:
+            torch = None
+
+        # Simulate the assignment logic in _ray_fit()
+        if assigned_gpu_ids:
+            os.environ['CUDA_VISIBLE_DEVICES'] = ','.join(map(str, assigned_gpu_ids))
+
+        # Report what the task actually sees
+        report = {
+            'task_id': task_id,
+            'assigned_gpu_ids': assigned_gpu_ids,
+            'CUDA_VISIBLE_DEVICES': os.environ.get('CUDA_VISIBLE_DEVICES', 'not set'),
+        }
+
+        if torch is not None:
+            report['torch_gpu_count'] = torch.cuda.device_count()
+            if torch.cuda.is_available():
+                report['torch_current_device'] = torch.cuda.current_device()
+            else:
+                report['torch_current_device'] = None
+        else:
+            report['torch_gpu_count'] = 'torch not available'
+            report['torch_current_device'] = 'torch not available'
+
+        return report
+
+    return ray_task_report_gpu_info
+
+
+class TestRayGpuAssignmentIntegration:
+    """Integration tests that verify GPU assignments work correctly with Ray."""
+
+    @staticmethod
+    def _get_system_gpu_count():
+        """Get actual GPU count on the system."""
+        try:
+            import torch
+            return torch.cuda.device_count()
+        except Exception:
+            return 0
+
+    def test_ray_gpu_assignment_no_gpu_integration(self):
+        """
+        Integration Test: Verify GPU assignment with no GPUs
+        This test should run on continuous_integration.yaml (single GPU CI)
+
+        Scenario: 0 GPUs, 1 task
+        Expected: Task runs on CPU, CUDA_VISIBLE_DEVICES = 'not set'
+        """
+        try:
+            import ray
+        except ImportError:
+            pytest.skip("Ray not available")
+
+        if not ray.is_initialized():
+            ray.init(num_cpus=4, num_gpus=0, ignore_reinit_error=True)
+
+        strategy = _construct_parallel_fold_strategy(num_cpus=4, num_gpus=0, num_jobs=1)
+        gpu_assignments = {}
+
+        gpu_assignments = strategy._calculate_gpu_assignment(
+            gpu_assignments=gpu_assignments,
+            task_id=0,
+            gpus_per_task=0,
+            total_gpus=0
+        )
+
+        assert gpu_assignments[0] == []
```

Comment:
we should verify more than just this from the output, such as len(gpu_assignments).